### ISSUE TITLE
Recusion in parse.js: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by reezer** [cheeriojs/cheerio#240]
On certain HTMLs I receive the following error.  Maybe connect() could be implemented without a recursion?

```
(...)/node_modules/cheerio/lib/parse.js:47
  _.each(dom, function(elem) {
                      ^
RangeError: Maximum call stack size exceeded
```

**Comment by josephrocca** [comment]
I've got this problem too :( hopefully the --stack-size option will be a good workaround for now.

Couldn't cheerio just keep track of the recursion count and if it gets too big, call the next recursion with `setTimeout(() => { recursiveFunction(...) }, 1)` to clear the stack?

**Comment by fb55** [comment]
Fixed by #985
